javasphinx/apidoc.py

```diff
@@ -192,8 +200,14 @@ def generate_from_source_file(doc_compiler, source_file, cache_dir):
     except Exception:
         util.unexpected('Unexpected exception while parsing %s', source_file)
 
+    documents = {}
     try:
-        documents = doc_compiler.compile(ast)
+        if source_file.endswith("package-info.java"):
+            for _, node in ast.filter(javalang.tree.PackageDeclaration):
+                document = doc_compiler.compile_package_documentation(node)
+                documents[node.name] = (node.name, node.name, document.build())
```

Comment:
I think I would prefer to use 'package-info' as the document name here,

```
documents[node.name] = (node.name, 'package-info', document.build())
```

This would make the implementation of `is_package_info_doc` more obvious I think and would also avoid any potential conflicts with actual class names.